File: frontend/src/api/news.ts
import client from './client'
import type { NewsItem } from '@/types'

export interface SearchParams {
  keyword?: string
  domain?: string
  city?: string
  source?: string
  min_importance?: number
  date_from?: string
  date_to?: string
  page?: number
  page_size?: number
}

export interface SearchResult {
  items: NewsItem[]
  total: number
  page: number
  page_size: number
}

export function searchNews(params: SearchParams): Promise<SearchResult> {
  return client.get('/news/search', { params }).then(r => (r as any).data)
}

export function getNewsById(id: string): Promise<NewsItem> {
  return client.get(`/news/${id}`).then(r => (r as any).data)
}

export function getNewsContent(id: string): Promise<{ id: string; title: string; content: string; word_count: number }> {
  return client.get(`/news/${encodeURIComponent(id)}/content`).then(r => (r as any).data)
}

export function fetchMeta(field: string): Promise<any[]> {
  return client.get(`/meta/${field}`).then(r => (r as any).data)
}

File: api/models/news.py
"""News search models."""
from datetime import date
from typing import Optional
from pydantic import BaseModel, Field


class NewsItem(BaseModel):
    id: str
    date: str
    source: str
    source_name: Optional[str] = ""
    source_url: Optional[str] = ""
    rank: Optional[int] = 0
    title: str
    domain: list[str] = []
    cities: list[str] = []
    entities: list[dict] = []
    tags: list[str] = []
    importance: int = 0
    ai_summary: Optional[str] = ""
    ai_why_matters: Optional[str] = ""
    file_path: Optional[str] = ""
    word_count: Optional[int] = 0


class NewsSearchParams(BaseModel):
    keyword: Optional[str] = None
    domain: Optional[str] = None
    city: Optional[str] = None
    source: Optional[str] = None
    min_importance: Optional[int] = Field(default=0, ge=0, le=5)
    date_from: Optional[str] = None
    date_to: Optional[str] = None
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)
    sort_by: str = Field(default="date", pattern="^(date|importance)$")
    sort_order: str = Field(default="desc", pattern="^(asc|desc)$")
